src/Workspaces/Core/Portable/Classification/SyntaxClassification/AbstractSyntaxClassificationService.Worker.cs

```diff
@@ -99,9 +100,15 @@ private void ProcessNodes()
                     _cancellationToken.ThrowIfCancellationRequested();
                     var nodeOrToken = _pendingNodes.Pop();
 
-                    if (nodeOrToken.FullSpan.IntersectsWith(_textSpan))
+                    if (_textSpanIntervalTree.HasIntervalThatIntersectsWith(nodeOrToken.FullSpan))
                     {
-                        ClassifyNodeOrToken(nodeOrToken);
+                        var intervalsThatIntersect = _textSpanIntervalTree.GetIntervalsThatIntersectWith(
+                            nodeOrToken.FullSpan.Start, nodeOrToken.FullSpan.Length);
+
+                        foreach (var interval in intervalsThatIntersect)
```

Comment:
do we want to call `ClassifyNodeOrToken` for all intervals or only first one is enough

Reply:
I would think all of them. One question I have here is whether you need to call both HasIntervalThatIntersectsWith and GetIntervalsThatIntersectWith. I would think you could just call GetIntervalsThatIntersectWith and if it's Length > 0, then go in and do the work